Packs/FireMonSecurityManager/Integrations/FireMonSecurityManager/FireMonSecurityManager.py

```diff
@@ -165,215 +130,319 @@ def create_pp_ticket(self, auth_token, payload):
+                    del result[i]["pcaResult"]["device"]["gpcImplementDate"]
+                if "state" in result[i]["pcaResult"]["device"]:
+                    del result[i]["pcaResult"]["device"]["state"]
+                if "managedType" in result[i]["pcaResult"]["device"]:
+                    del result[i]["pcaResult"]["device"]["managedType"]
+                if "gpcStatus" in result[i]["pcaResult"]["device"]:
+                    del result[i]["pcaResult"]["device"]["gpcStatus"]
+                if "updateMemberRuleDoc" in result[i]["pcaResult"]["device"]:
+                    del result[i]["pcaResult"]["device"]["updateMemberRuleDoc"]
+                if "devicePack" in result[i]["pcaResult"]["device"]:
+                    del result[i]["pcaResult"]["device"]["devicePack"]
+
+    return CommandResults(
+        outputs_prefix="FireMonSecurityManager.PCA",
+        outputs_key_field="pca",
+        outputs=result,
+        readable_output=tableToMarkdown(
+            name="FireMon SecurityManager PCA:",
+            t=result[0]["pcaResult"]["preChangeAssessmentControls"],
+            removeNull=True,
+        ),
+        raw_response=list_of_device_changes,
+    )
+
+
+def get_paged_search_secrule(client: Client, auth_token: str, payload: Dict[str, Any]) -> List:
+    """Make subsequent requests using client and other arguments
+
+    Args:
+        client (Client): `Client` class object
+        auth_token (str): authentication token to use
+        payload (Dict[str, Any]): parameter payload to use
+
+    Returns:
+        (List[Dict[str, Any]]): results list
+    """
+    result = list()
+    response = client.get_paged_search_secrule(auth_token, payload)
+    total_pages = response.get("total", 0) // payload.get("pageSize")
+
+    result.extend(response.get("results", list()))
+
+    while payload.get("page") < total_pages:  # NOTE: Check if we can implement async here
+        payload["page"] += 1
+        response = client.get_paged_search_secrule(auth_token, payload)
+        result.extend(response.get("results", list()))
+
+    return result
+
+
+def secmgr_secrule_search_command(client: Client, args: Dict[str, Any]):
+    """Searches for security rules using the SIQL language query
+
+    Args:
+        client (Client): `Client` class object
+        args (Dict[str, Any]): demisto arguments passed
+    """
+    auth_token_cmd_result = authenticate_command(client)
+    auth_token = auth_token_cmd_result.outputs
+
+    # page size can't be less than 1
+    page_size = 1 if int(args.get("pageSize", 10)) < 1 else int(args.get("pageSize", 10))
+    payload = dict(
+        q=str(args.get("q")),
+        pageSize=page_size,
+        page=int(args.get("page", 0)),
+    )
+
+    try:
+        results = get_paged_search_secrule(client, auth_token, payload)
+    except DemistoException as err:
+        demisto.error(f"{err}")
+        msg = None if err.res is None else err.res.json().get("message")
+        raise DemistoException(msg)
```

Comment:
it does not have to be framed with try and except in the command function itself, if an error returns it should be handled in the main function

Reply:
@merit-maita , I have removed try/catch from the command function, as it is already handled in the main function try/catch block so don't think anything new is required.

I haven't added any new file except a `get_paged_search_secrule.json` in test_data for the test case. I'm not sure but maybe those files were added when this PR was raised from the marketplace. 

I'm working from the IST timezone, please let me know what time suits you for the demo.